Treasure hunter

Codidodido opened the pull request.

Codidodido committed:
Initialize treasure hunt project

Codidodido committed:
Create map class

- Define map properties
- Initialize game map
- Find treasure position
- Find trap position
- Display map

Codidodido committed:
Create player class

- Define player properties

Codidodido committed:
Edit player class script

Implement SetRandomPosition method which sets random position for player

Codidodido committed:
Edit map class script

Implement OptimizeMap method which optimize map and remove repeated max values or min values

Codidodido committed:
Edit program class script

Create a map and player and StartGame method which starts game

Codidodido committed:
Edit program class script

Implement game logics and inputs

Codidodido committed:
Commit extra files

-> exe file
-> pdbfile
-> csproj file

URAmiRBin commented on Algorithms and Data Structures/TreasureHunt/TreasureHunt/Map.cs:
```diff
@@ -0,0 +1,149 @@
+                    if (_map[i, j] > maxValue)
+                    {
+                        maxValue = _map[i, j];
+                        maxValueIndex = new[] { i, j };
```
why new everytime?

URAmiRBin commented on Algorithms and Data Structures/TreasureHunt/TreasureHunt/Map.cs:
```diff
@@ -0,0 +1,149 @@
+            }
+        }
+
+        private void OptimizeMap()
+        {
+            for (var i = 0; i < _mapX; i++)
+            {
+                for (var j = 0; j < _mapY; j++)
+                {
+                    if (_map[i, j] == _maxValue && i != TreasureX && j != TreasureY)
+                    {
+                        _map[i, j]--;
+                    }
+                    
+                    if (_map[i, j] == _minValue && i != TrapX && j != TrapY)
+                    {
+                        _map[i, j]++;
+                    }
+                }
+            }
+        }
+    }
```
we do optimization when we need to, we dont need any here.

keep in mind that we call it optimization when it really optimize the calculation, this just adds some useless calculation to clean the data that adds to the load.

also these sorts of stuff will make the reader more confused.

URAmiRBin commented:
@Codidodido 
We do not merge to `medrick-bridge-tech/Exercises:main` change it to you own main branch so you can merge it

Codidodido committed:
Refactor map script

-> Assign i and j directly instead of new it

Codidodido committed:
Edit map script

- Remove optimize method
- Implement an algorithm to place max value and min value after initlaizing map

Codidodido committed:
Edit program script

define max and min value and assign it to map

Codidodido committed:
Commit new exe file

Codidodido committed:
Edit map script

-> Make initialize algorithm simple

Codidodido committed:
Commit new exe
